tests/devices/test_default_qutrit_mixed.py

```diff
@@ -12,29 +12,1004 @@
+    """Test that the device behaves correctly when provided with a random seed."""
+
+    measurements = [
+        [qml.sample(wires=0)],
+        [qml.expval(qml.GellMann(0, 3))],
+        [qml.counts(wires=0)],
+        [qml.sample(wires=0), qml.expval(qml.GellMann(0, 8)), qml.counts(wires=0)],
+    ]
+
+    @pytest.mark.parametrize("measurements", measurements)
+    def test_same_seed(self, measurements):
+        """Test that different devices given the same random seed will produce
+        the same results."""
+        qs = qml.tape.QuantumScript([qml.THadamard(0)], measurements, shots=1000)
+
+        dev1 = DefaultQutritMixed(seed=123)
+        result1 = dev1.execute(qs)
+
+        dev2 = DefaultQutritMixed(seed=123)
+        result2 = dev2.execute(qs)
+
+        if len(measurements) == 1:
+            result1, result2 = [result1], [result2]
+
+        assert all(np.all(res1 == res2) for res1, res2 in zip(result1, result2))
+
+    @pytest.mark.slow
+    def test_different_seed(self):
+        """Test that different devices given different random seeds will produce
+        different results (with almost certainty)."""
+        qs = qml.tape.QuantumScript([qml.THadamard(0)], [qml.sample(wires=0)], shots=1000)
+
+        dev1 = DefaultQutritMixed(seed=None)
+        result1 = dev1.execute(qs)
+
+        dev2 = DefaultQutritMixed(seed=123)
+        result2 = dev2.execute(qs)
+
+        dev3 = DefaultQutritMixed(seed=456)
+        result3 = dev3.execute(qs)
+
+        # assert results are pairwise different
+        assert np.any(result1 != result2)
+        assert np.any(result1 != result3)
+        assert np.any(result2 != result3)
+
+    @pytest.mark.parametrize("measurements", measurements)
+    def test_different_executions(self, measurements):
+        """Test that the same device will produce different results every execution."""
+        qs = qml.tape.QuantumScript([qml.THadamard(0)], measurements, shots=1000)
+
+        dev = DefaultQutritMixed(seed=123)
+        result1 = dev.execute(qs)
+        result2 = dev.execute(qs)
+
+        if len(measurements) == 1:
+            result1, result2 = [result1], [result2]
+
+        assert all(np.any(res1 != res2) for res1, res2 in zip(result1, result2))
+
+    @pytest.mark.parametrize("measurements", measurements)
+    def test_global_seed_and_device_seed(self, measurements):
+        """Test that a global seed does not affect the result of devices
+        provided with a seed."""
+        qs = qml.tape.QuantumScript([qml.THadamard(0)], measurements, shots=1000)
+
+        # expected result
+        dev1 = DefaultQutritMixed(seed=123)
+        result1 = dev1.execute(qs)
+
+        # set a global seed both before initialization of the
+        # device and before execution of the tape
+        np.random.seed(456)
+        dev2 = DefaultQutritMixed(seed=123)
+        np.random.seed(789)
+        result2 = dev2.execute(qs)
```

Comment:
@albi3ro we want to avoid changing the global seed in tests now right?

Reply:
I had one idea i can get the state of the rng change the rng then revert to the save state. That or I can set it to none at the end of the test.

Reply:
I undid that because it shows up in a few tests that change global seed.
These tests are similar to the qubit ones and I checked that they are still the same for the qubit version, do I need to change anything?

Reply:
Yeah I just checked the `default.qubit` tests and they do the same thing. No changes needed in that acse. Resolving.